src/acl.c

```diff
@@ -1422,6 +1433,26 @@ int ACLAuthenticateUser(client *c, robj *username, robj *password) {
     }
 }
 
+/* Attempt authenticating the user. First through Module based custom authentication if any exists,
+ * and otherwise password based authentication.
+ * Returns C_OK if module based auth succeeded OR if in progress through a blocking custom auth
+ * OR if password based auth succeeded.
+ * Otherwise C_ERR is returned and errno is set to:
+ *  EINVAL: if the user is disabled or if the username-password do not match (password based auth).
+ *  ENONENT: if the specified user does not exist at all.
```

Comment:
ERRNO codes are set through inner function calls in `ACLCheckUserCredentials`.
``` 
moduleAuthentication  -> passwordBasedAuth -> ACLCheckUserCredentials
```

But because this is not directly done inside this (`ACLAuthenticationUser`), I removed it as suggested.
https://github.com/redis/redis/pull/11659/commits/92115fcc1fae2927656885c0e9b0c39ae9b61943